server/pulp/server/webservices/views/repo_groups.py

```diff
@@ -0,0 +1,370 @@
+        :param distributor_id: distributor to get
+        :type  distributor_id: str
+
+        :return: response containing a serialized dict of the added distributor
+        :rtype: django.http.HttpResponse
+        """
+        distributor_manager = managers_factory.repo_group_distributor_manager()
+        dist = distributor_manager.get_distributor(repo_group_id, distributor_id)
+        dist['_href'] = request.get_full_path()
+        return generate_json_response_with_pulp_encoder(dist)
+
+    @auth_required(authorization.DELETE)
+    @json_body_allow_empty
+    def delete(self, request, repo_group_id, distributor_id):
+        """
+        Disassociate a distributor from a repository group.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_group_id: repo group the distributor is associated with
+        :type  repo_group_id: str
+        :param distributor_id: distributor to disassociate
+        :type  distributor_id: str
+
+        :return: response containing None
+        :rtype: django.http.HttpResponse
+        """
+        params = request.body_as_json
+        force = params.get('force', False)
+        distributor_manager = managers_factory.repo_group_distributor_manager()
+        distributor_manager.remove_distributor(repo_group_id, distributor_id, force=force)
+        return generate_json_response(None)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_allow_empty
+    def put(self, request, repo_group_id, distributor_id):
+        """
+        Change information about a single repo group distributor association.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_group_id: repo group the distributor is associated with
+        :type  repo_group_id: str
+        :param distributor_id: distributor to update
+        :type  distributor_id: str
+
+        :return: response containing a serialized dict of the modified distributor association
+        :rtype: django.http.HttpResponse
+        :raises MissingValue: if required param 'distributor_config' is not in the body
+        """
+        params = request.body_as_json
+        distributor_config = params.get('distributor_config', None)
+        if distributor_config is None:
+            raise pulp_exceptions.MissingValue(['distributor_config'])
+        distributor_manager = managers_factory.repo_group_distributor_manager()
+        result = distributor_manager.update_distributor_config(repo_group_id, distributor_id,
+                                                               distributor_config)
+        result['_href'] = request.get_full_path()
+        return generate_json_response_with_pulp_encoder(result)
+
+
+class RepoGroupPublishView(View):
+    """
+    View to trigger the publish of a repo group.
+    """
+
+    @auth_required(authorization.EXECUTE)
+    @json_body_allow_empty
```

Comment:
In order to maintain the current API, I let an empty body pass through the decorator so it can be handled by the view. In this case, it will be caught by the logic of the view and will raise a `MissingValue` with a 400. Using `json_body_required` would also return a 400, but with a different error.